pkg/ebda/ebda.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package ebda looks for the EBDA pointer in /dev/mem, and provides access to the EBDA.
+// This is useful for us to read or write to the EBDA, for example to copy the RSDP
+// into the EBDA.
+package ebda
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+)
+
+const (
+	// EBDAAddressOffset is the traditional offset where the location of the EBDA is stored.
+	EBDAAddressOffset = 0x40E
+)
+
+// EBDA represents an EBDA region in memory
+type EBDA struct {
+	BaseOffset int64
+	Length     int64
```

Comment:
From my understanding, `Length` always equal `len(Buf)`, so you can get rid of this field.

Reply:
I'll let people do what they want at their own risk